src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -446,35 +446,54 @@ impl Step for RustAnalyzer {
 
     /// Runs `cargo test` for rust-analyzer
     fn run(self, builder: &Builder<'_>) {
-        let host = self.compilers.target();
+        let tool_result = builder.ensure(tool::RustAnalyzer::from_compilers(self.compilers));
+        let build_compiler = tool_result.build_compiler;
+        let target = self.compilers.target();
+
+        let test_compiler = self.compilers.target();
```

Comment:
```suggestion
        let target = self.compilers.target();
```
The variable is duplicated now.